### PR TITLE
contributing: add guidance on how to update package versions support

### PR DESCRIPTION
**Pull request by emdneto**
# Description
Improve contributing guide to make explicit how to support new versions of instrumentation packages

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -278,6 +278,19 @@ Below is a checklist of things to be mindful of when implementing a new instrume
...
+### Update supported instrumentation package versions
+
+- Update the respective instrumentation `pyproject.toml` file in _instruments_ entry under `[project.optional-dependencies]` section
+- Update instrumentation `package.py` file in `_instruments` variable
+- Run command `tox -e generate` to regenerate necessary files
+
+If you're adding support for a new version of the instrumentation package, follow these additional steps:
+
+- Add new test-requirements.txt file with the respective package version required for testing
+- Add a new test environment entry for the package version in `tox.ini` and run `tox -e generate-workflows` to generate new workflows
```
Comment (codefromthecrypt) [inline CONTRIBUTING.md:290]:
```suggestion
- Add a new test environment entry for the package version in [tox.ini](tox.ini) and run `tox -e generate-workflows` to generate new workflows
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -278,6 +278,19 @@ Below is a checklist of things to be mindful of when implementing a new instrume
 - Testing
   - When adding a new instrumentation remember to update `tox.ini` adding appropriate rules in `envlist`, `command_pre` and `commands` sections
 
+### Update supported instrumentation package versions
+
+- Update the respective instrumentation `pyproject.toml` file in _instruments_ entry under `[project.optional-dependencies]` section
```
Comment (codefromthecrypt) [inline CONTRIBUTING.md:283]: maybe file the first two under a bullet for "In the project directory" and the last I think needs to be "In the repository root" right?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -278,6 +278,19 @@ Below is a checklist of things to be mindful of when implementing a new instrume
...
+- Update instrumentation `package.py` file in `_instruments` variable
+- Run command `tox -e generate` to regenerate necessary files
+
+If you're adding support for a new version of the instrumentation package, follow these additional steps:
+
+- Add new test-requirements.txt file with the respective package version required for testing
```
Comment (codefromthecrypt) [inline CONTRIBUTING.md:289]: similarly, I think it helps to clarify the first step is in the package directory and the second from the repository root.

**Review by codefromthecrypt — COMMENTED** [review]
Looks good, I have some suggestions to clarify the location of where commands should run. Also here and everywhere else in this file, we should switch from backticking tox.ini to a link to it. This helps markdown link checker ensure there's not a typo there, also, readers can click it!

File: CONTRIBUTING.md (hunk 2 above)
Comment (xrmx) [reply CONTRIBUTING.md:283]: Yes, first two are inside instrumentation package directory.

File: CONTRIBUTING.md (hunk 1 above)
Comment (emdneto) [reply CONTRIBUTING.md]: Thanks! Updated it accordingly

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -278,6 +278,20 @@ Below is a checklist of things to be mindful of when implementing a new instrume
 - Testing
   - When adding a new instrumentation remember to update `tox.ini` adding appropriate rules in `envlist`, `command_pre` and `commands` sections
 
+### Update supported instrumentation package versions
+
+- Navigate to the **instrumentation package directory:**
+  - Update the respective instrumentation **`pyproject.toml`** file by modifying _instruments_ entry in the `[project.optional-dependencies]` section with the new version constraint
+  - Update instrumentation **`package.py`** file in `_instruments` variable with the new version constraint
```
Comment (xrmx) [inline CONTRIBUTING.md:285]:
```suggestion
  - Update `_instruments` variable in instrumentation **`package.py`** file with the new version constraint
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -278,6 +278,20 @@ Below is a checklist of things to be mindful of when implementing a new instrume
 - Testing
   - When adding a new instrumentation remember to update `tox.ini` adding appropriate rules in `envlist`, `command_pre` and `commands` sections
 
+### Update supported instrumentation package versions
+
+- Navigate to the **instrumentation package directory:**
+  - Update the respective instrumentation **`pyproject.toml`** file by modifying _instruments_ entry in the `[project.optional-dependencies]` section with the new version constraint
```
Comment (xrmx) [inline CONTRIBUTING.md:284]:
```suggestion
  - Update **`pyproject.toml`** file by modifying _instruments_ entry in the `[project.optional-dependencies]` section with the new version constraint
```

**Review by xrmx — APPROVED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -89,7 +89,9 @@ for more detail on available tox commands.
 
 ### Troubleshooting
 
-> Some packages may require additional system wide dependencies to be installed. For example, you may need to install `libpq-dev` to run the postgresql client libraries instrumentation tests. or `libsnappy-dev` to run the prometheus exporter tests. If you encounter a build error, please check the installation instructions for the package you are trying to run tests for.
+Some packages may require additional system-wide dependencies to be installed. For example, you may need to install `libpq-dev` to run the postgresql client libraries instrumentation tests or `libsnappy-dev` to run the prometheus exporter tests. If you encounter a build error, please check the installation instructions for the package you are trying to run tests for. 
+
+For `docs` building, you may need to install `mysql-client` and other required dependencies as necessary. Ensure the Python version used in your local setup matches the version used in the CI to maintain compatibility when building the documentation.
```
Comment (emdneto) [inline CONTRIBUTING.md:94]: Added a note about docs instructions

Comment (tammy-baylis-swi) [inline CONTRIBUTING.md:94]: I bumped into this the other day and figured it out by searching Slack and finding a thread that points to [this line](https://github.com/open-telemetry/opentelemetry-python/blob/main/.github/workflows/misc_0.yml#L121). Could this paragraph include a hyperlink to maybe the [workflows dir](https://github.com/open-telemetry/opentelemetry-python/tree/main/.github/workflows)?

**Review by tammy-baylis-swi — APPROVED** [review]

**Review by lzchen — APPROVED** [review]

File: CONTRIBUTING.md (hunk 6 above)
Comment (emdneto) [reply CONTRIBUTING.md:94]: Sure. Will add

Comment (emdneto) [inline CONTRIBUTING.md:94]:
```suggestion
For `docs` building, you may need to install `mysql-client` and other required dependencies as necessary. Ensure the Python version used in your local setup matches the version used in the [CI](./.github/workflows/) to maintain compatibility when building the documentation.
```

**Pull request merged**